src/controller/src/clusters.rs

```diff
@@ -547,28 +547,40 @@ where
     }
 
     pub fn events_stream(&self) -> BoxStream<'static, ClusterEvent> {
-        fn translate_event(event: ServiceEvent) -> Result<ClusterEvent, anyhow::Error> {
+        let deploy_generation = self.deploy_generation;
+
+        fn translate_event(event: ServiceEvent) -> Result<(ClusterEvent, u64), anyhow::Error> {
```

Comment:
maybe a bit brittle that we're passing around a bare `u64` here instead of a typed generation version, but ... 🤷‍♂️